### PR TITLE
Convert circleci workflows to github actions

### PR DESCRIPTION
**Pull request by robandpdx**
This pull request converts the CircleCI workflows to GitHub actions workflows.  

`build-deb_oldoldstable` job fails with the following error.
```
/bin/bash: dx: command not found
```

[Here is the latest workflow run in my fork](https://github.com/robandpdx-org/redex/actions/runs/8351358902).

---
https://fburl.com/workplace/f6mz6tmw

### CODE REVIEW

**Comment by thezhangwei** [comment]
Thanks for putting up the change. It looks like the remaining issue is that the `dx` binary cannot be found for one of the build?

File: .github/workflows/build_main.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    - uses: "./.github/actions/build_debian"
+      with:
+        job_name: ubuntu_22.04
+  build-deb_oldoldstable:
+    runs-on: 4-core-ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v4.1.1
+    - uses: "./.github/actions/build_debian"
+      with:
+        use_sdk: false
```
Comment (wsanville) [inline .github/workflows/build_main.yml:35]: General question: what is build-deb_oldoldstable?

Also, is the `dx: command not found` error related to use_sdk being set to false here? What happens if it's true?

**Review by wsanville — COMMENTED** [review]
See question inline.

**Comment by thezhangwei** [comment]
Hey @robandpdx how can we kick off the github action run on a PR if we were to iterate on top of this change?

File: .github/workflows/build_main.yml (hunk 1 above)
Comment (robandpdx) [reply .github/workflows/build_main.yml:35]: `build-deb_oldoldstable` is meant to mirror the [circleci job here](https://github.com/facebook/redex/blob/main/.circleci/config.yml#L478). It specifies `use_sdk: false` as does the circleci job.

Honestly, I don't know what happens if we change this to `true`. I'm hoping someone with better knowledge of this project can pick up where I left of and get this over the finish line.

**Comment by robandpdx** [comment]
> Hey @robandpdx how can we kick off the github action run on a PR if we were to iterate on top of this change?

@thezhangwei I have been working in a fork and merging changes to main in my fork in order to trigger the workflows.

**Comment by bigfootjon** [comment]
https://github.com/facebook/redex/commit/3e74d9319b0c2479c867b86a0b046e7294fd8ac6 did this instead

**Pull request closed without merging**